### PR TITLE
fix: ontology and ontology_array input for showing existing value

### LINKED ISSUE #1008: Filled out data in ontology and ontology_array data types is not visible when editing in forms
**What version of EMX2 are you using (see footer)**
8.54.0

**Describe the bug**

**To Reproduce**
1. Go to  https://emx2.dev.molgenis.org/Conception-test/tables/#/Cohorts
2. See that e.g. countries, and type are filled out for cohort called 'test' if you scroll to the right in the table shown
3. Try to edit this cohort in forms: these fields are not visibly filled out in the form, as if no data is present.

### PR DESCRIPTION
**Pull request by mswertz**
closes #1008

### CODE REVIEW

File: apps/styleguide/src/forms/InputOntology.vue [hunk 1]
```diff
@@ -283,22 +287,32 @@ export default {
         });
       } else {
         //non-list, deselect all
-        this.terms.forEach((term) => (term.selected = false));
+        Object.keys(this.terms).forEach(
+          (key) => (this.terms[key].selected = false)
+        );
+      }
```
Comment (connoratrug) [inline apps/styleguide/src/forms/InputOntology.vue:292]: code duplication , can be moved outside the if

File: apps/styleguide/src/forms/InputOntology.vue [hunk 2]
```diff
@@ -336,7 +388,7 @@ export default {
         });
       }
       //collapse all first
-      Object.values(this.terms).forEach((t) => (t.expanded = false));
+      //Object.values(this.terms).forEach((t) => (t.expanded = false));
```
Comment (connoratrug) [inline apps/styleguide/src/forms/InputOntology.vue:390]: remove unused code

**Comment by mswertz** [comment]
[ontology input bug](https://trello.com/c/IIwjz0P1/419-ontology-input-bug)

**Review by connoratrug — COMMENTED** [review]
when i search , the search filter hides the (previously) selected items . This seems like a bug to me but maybe it is the desired behaviour ?

https://user-images.githubusercontent.com/47183404/157839708-3738c925-4cb3-4f72-a96c-b84a55dbdb41.mov

**Review by connoratrug — APPROVED** [review]
not use on some of the implementation details , bug the pr fixes the reported issue

File: apps/styleguide/src/forms/InputOntology.vue (hunk 1 above)
Comment (mswertz) [reply apps/styleguide/src/forms/InputOntology.vue:292]: the duplicate is actually in another method

File: apps/styleguide/src/forms/InputOntology.vue (hunk 2 above)
Comment (mswertz) [reply apps/styleguide/src/forms/InputOntology.vue:390]: agreed

File: apps/styleguide/src/forms/InputOntology.vue (hunk 1 above)
Comment (mswertz) [reply apps/styleguide/src/forms/InputOntology.vue:292]: I don't think so? the duplicate is in another function 'select'

File: apps/styleguide/src/forms/InputOntology.vue (hunk 2 above)
Comment (mswertz) [reply apps/styleguide/src/forms/InputOntology.vue:390]: indeed

**Comment by mswertz** [comment]
> when i search , the search filter hides the (previously) selected items . This seems like a bug to me but maybe it is the desired behaviour ?
> 
>  Screen.Recording.2022-03-11.at.10.24.03.mov

Indeed by design. But I was on the fence on this one, but I thought we probably get more confusion if we would show all selected even if not matching searches.

**Pull request merged**